### PR TITLE
[CN-602] Add existing CR validation to validating webhook

### PR DESCRIPTION
**Pull request by dzeromski-hazelcast**
## Description

Add validating webhook for Hazlecast cluster and Topic data structure CRs. 

## User Impact

This change allows operator to report issues with CRs before they are applied.

### CODE REVIEW

**Review by mtyazici — APPROVED** [review]

File: api/v1alpha1/map_validation.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
+package v1alpha1
+
+import (
+	"errors"
+)
+
+func ValidateNotUpdatableFields(current *MapSpec, last *MapSpec) error {
```
Comment (SeriyBg) [inline api/v1alpha1/map_validation.go:7]: What about the validation for the other CRs?

Comment (dzeromski-hazelcast) [reply api/v1alpha1/map_validation.go:7]: We will add as soon as we have it. This PR just adds existing validation do webhook where applicable.

Comment (SeriyBg) [reply api/v1alpha1/map_validation.go:7]: We do have the validation for the `Cache CR` with persistence 😃

Comment (dzeromski-hazelcast) [reply api/v1alpha1/map_validation.go:7]: Sure we can add it in https://github.com/hazelcast/hazelcast-platform-operator/pull/499 where we add Webhooks for cache :)

**Review by SeriyBg — APPROVED** [review]

**Pull request merged**